test/MLOps.NET.AWS.Tests/S3BucketModelRepositoryTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Amazon;
+using Amazon.Runtime;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Amazon.S3.Util;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MLOps.NET.Storage;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.IO.Abstractions.TestingHelpers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MLOps.NET.AWS.Tests
+{
+    [TestClass]
+    public class S3BucketModelRepositoryTests
+    {
+        [TestMethod]
+        public async Task UploadModelAsync_ShouldSaveFileInS3Bucket()
```

Comment:
In this particular test, can we verify that not only the Put method is called, but also that it's called with the correct arguments? E.g. bucket name, run id, file type etc? That would be awesome! You can do that by using 
```
It.Is<PutObjectRequest>(x => x.BucketName == ""....)
```

Reply:
Nice one . Thanks @aslotte